fdbserver/TLogServer.actor.cpp

```diff
@@ -1817,26 +1817,23 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	// if tLog locked for recovery, return an empty message at the cluster recovery version
 	// if requested version is greater than any received.
 	state Optional<Version> clusterRecoveryVersion = Optional<Version>();
-	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR && logData->stopped()) {
-		if (!self->clusterRecoveryVersion.get().present()) {
-			TraceEvent("TLogPeekMessages0", self->dbgid)
-				.detail("LogId", logData->logId)
-				.detail("ReqReturnIfBlocked", reqReturnIfBlocked ? 1 : 0)
-				.detail("ReqBegin", reqBegin)
-				.detail("Version", logData->version.get())
-				.detail("RecoveredAt", logData->recoveredAt);
-			if (logData->version.get() < reqBegin) {
-				TraceEvent("NoClusterRecoveryVersion",self->dbgid).detail("X",reqReturnIfBlocked);
-				wait(self->clusterRecoveryVersion.onChange());
-				// wait(timeout(self->clusterRecoveryVersion.onChange(),10,Void()));
-				// ASSERT(self->clusterRecoveryVersion.get().present());
-				TraceEvent("YesClusterRecoveryVersion",self->dbgid);
-			}
+	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && logData->stopped()) {
+		DebugLogTraceEvent("TLogPeekMessagesWaitForCrv", self->dbgid)
+		    .detail("Tag", reqTag.toString())
+		    .detail("LogId", logData->logId)
+		    .detail("ReqReturnIfBlocked", reqReturnIfBlocked ? 1 : 0)
+		    .detail("ReqBegin", reqBegin)
+		    .detail("Version", logData->version.get())
+		    .detail("RecoveredAt", logData->recoveredAt);
+
+		if (!logData->clusterRecoveryVersion.get().present()) {
+			wait(logData->clusterRecoveryVersion.onChange());
 		}
-	}
-	if (logData->stopped() && logData->version.get() < reqBegin && self->clusterRecoveryVersion.get().present()) {
-		clusterRecoveryVersion = self->clusterRecoveryVersion.get().get();
-		TraceEvent("TLogPeekMessagesClusterRecoveryVersion", self->dbgid).detail("Version", clusterRecoveryVersion.get());
+		clusterRecoveryVersion = logData->clusterRecoveryVersion.get().get() + 1;
```

Comment:
We are bumping up the recovery version here - do we understand why this is really needed (I mean, do we understand why the logs try to peek till the bumped up version)?

Reply:
I am removing the +1 in this draft PR. But we still add one to the RV calculated in `endEpoch()`. We do this by sending the result of `TagPartitionedLogSystem::getEnd()` to the tLogs. This is what is expected by the cursors. Recall the cursors peek a range (x..y] , where the lower bound x is inclusive and the upper bound y is exclusive.

Reply:
I will add a comment so readers do not get thrown off by this.